### ISSUE TITLE
Computation state lost when navigating

### ISSUE THREAD

**Issue by stevejpurves** [executablebooks/myst-theme#223]
### Describe the bug

**context**
When I do start computation with `thebe` and navigate between pages.

**expectation**
I expected to be able to return to a page and still see it in the sate I left it in terms of thebe computation. i.e. I should be able to enable thebe and start a notebook or figure executing on one page, then navigate away and return and still see the correct state on my return, for example the page would still be executing or execution will have completed.

**bug**
But instead the state of the page and the state of the ExecutionScopeProvider is always reset on navigation. THis was not previously the case, and the current design should allow for navigation. This was previously working.

**problem**
This is a problem for people working with `thebe` as they cannot navigate between pages and maintain the state they computed.

### Reproduce the bug

Run examples at https://rowanc1.github.io/myst-lite/

**Comment by rowanc1** [comment]
Note that this never worked on staticly rendered pages. All GitHub pages are staticly rendered!

**Comment by stevejpurves** [comment]
Also this works on pages except for the "root" or "index" of the site, which I think is being fully re-rendered on navigation, akin to a page reload?

If that is the same with github pages then that points to needing a different way to manage the site state and keep any state for active pages / sessions alive.

Perhaps this is fixable for the non gh-pages builds. If so then for gh-pages, sounds like we'll need a different static strategy.

**Comment by rowanc1** [comment]
Yes, we could bring the state/providers up to root (not on an articlepage), this would fix the root/index page issue. Right now they are not shared between the `$.tsx` route and `_index.tsx`.
